bin/mix.ps1

```diff
@@ -1,23 +1,20 @@
-# Store path to mix.bat as a FileInfo object
-$mixBatPath = (Get-ChildItem (((Get-ChildItem $MyInvocation.MyCommand.Path).Directory.FullName) + '\mix.bat'))
-$newArgs = @()
+#!/usr/bin/env pwsh
 
-for ($i = 0; $i -lt $args.length; $i++)
-{
-  if ($args[$i] -is [array])
-  {
-    # Commas created the array so we need to reintroduce those commas
-    for ($j = 0; $j -lt $args[$i].length - 1; $j++)
-    {
-      $newArgs += ($args[$i][$j] + ',')
-    }
-    $newArgs += $args[$i][-1]
+$ScriptPath = Split-Path -Parent $PSCommandPath
+$Command = Join-Path -Path $ScriptPath -ChildPath "elixir.ps1"
+$MixCommand = Join-Path -Path $ScriptPath -ChildPath "mix"
+
+$NewArgs = @($MixCommand)
+
+for ($i = 0; $i -lt $Args.Count; $i++) {
+  $Arg = $Args[$i]
+
+  if ($Arg -is [string]) {
+    $NewArgs += $Arg
   }
-  else
-  {
-    $newArgs += $args[$i]
+  else {
+    $NewArgs += [string]::Join(",", $Arg)
```

Comment:
should keep the original comment as it is not obvious why an argument would be an array.

Reply:
Since I'm sourcing the main file and not changing the args for that, we don't need this comment anymore.